src/System.Net.HttpListener/tests/HttpListenerRequestTests.cs

```diff
@@ -82,7 +82,24 @@ public static IEnumerable<object[]> ContentEncoding_TestData()
             yield return new object[] { "Content-Type:application.json;charset=,", Encoding.Default };
             yield return new object[] { "Content-Type:application.json;charset=\"unicode", Encoding.Default };
             yield return new object[] { "Content-Type:application/json;charset=NoSuchEncoding", Encoding.Default };
-            yield return new object[] { "Content-Type:application/json;charset=unicode; boundary=something", Encoding.Default };
```

Comment:
The test data is not correct and I changed it.

Reason:
Without the change, the `charset=unicode;` will be parsed as `unicode;` (notice the semicolon inside the value), thus the Encoding.GetEncoding(charSet) will fail with ArgumentException. In this case, Encoding.Default will be chose for ContentEncoding, even if client explicitly specifies the Unicode encoding in the header.